src/commands/list_folder.ts

```diff
@@ -46,7 +46,9 @@ new CLICommand({
 				CLI_APP_NAME,
 				CLI_APP_VERSION
 			);
-			children = await arDrive.listPrivateFolder(folderId, password);
+			const driveId = await arDrive.getDriveIdForFolderId(folderId);
```

Comment:
This fires one extra query to get the `driveID` for the specified folder

Reply:
Is there a way for the user to provide the driveKey rather than their password?

Reply:
There is. I'll mention you at that line

Reply:
Thats true there, but we only enter this `if (wallet && password)`. So when I provide a wallet && driveKey, we'd attempt the public version here

Reply:
Got ya! I didn't realize that, here the change: 60932cf33681a2b7e1f35043ec96c7d6462908fd